spras/local_neighborhood.py

```diff
@@ -0,0 +1,163 @@
+import warnings
+from pathlib import Path
+import pandas as pd
+
+# Import necessary utilities from the spras package
+from spras.containers import prepare_volume, run_container_and_log
+from spras.interactome import reinsert_direction_col_undirected
+from spras.prm import PRM
+from spras.util import add_rank_column, duplicate_edges, raw_pathway_df
+
+__all__ = ['LocalNeighborhood']
+
+class LocalNeighborhood(PRM):
+    """
+    Local Neighborhood Algorithm Wrapper
+
+    This class wraps the Local Neighborhood algorithm for use within the SPRAS framework.
+    It handles generating the required input files (network and nodes) and running
+    the algorithm inside a Docker container.
+
+    Expected raw input network format (for --network):
+    vertex1|vertex2
+    - Each line represents an interaction. No header.
+    - Assumed to be undirected or treated as such by the algorithm.
+
+    Expected raw input nodes file format (for --nodes):
+    NodeID
+    - A simple list of node IDs, one per line. These are the query nodes for the algorithm.
+    """
+    required_inputs = ['network', 'nodes']
+
+    @staticmethod
+    def generate_inputs(data, filename_map):
+        for input_type in LocalNeighborhood.required_inputs:
+            if input_type not in filename_map:
+                raise ValueError(f"{input_type} filename is missing in filename_map")
+
+        node_df = data.node_table
+
+        has_prize = node_df['prize'].notna() & (node_df['prize'] != '')
+        is_active = node_df['active'] == True if 'active' in node_df.columns else pd.Series(False, index=node_df.index)
+        is_source = node_df['sources'] == True if 'sources' in node_df.columns else pd.Series(False, index=node_df.index)
+        is_target = node_df['targets'] == True if 'targets' in node_df.columns else pd.Series(False, index=node_df.index)
+
+        selected_nodes_series = node_df.loc[has_prize | is_active | is_source | is_target, 'NODEID']
+
+        if selected_nodes_series.empty:
+            raise ValueError("No relevant nodes found to create the 'nodes' input file for Local Neighborhood.")
+
+        nodes_filepath = filename_map["nodes"]
+        with open(nodes_filepath, 'w') as f:
+            for node_id in sorted(selected_nodes_series.unique().tolist()):
```

Comment:
```suggestion
            for node_id in sorted(selected_nodes_series.unique()):
```
`sorted` takes any iterable, so you should be able to remove `.tolist()`